### ISSUE TITLE
serverless-offline-sqs: Incompatible with latest serverless-offline. Can't have certain variables outside handler

### ISSUE THREAD

**Issue by zoellner** [CoorpAcademy/serverless-plugins#143]
I'm running into issues using the latest versions of serverless-offline and serverless-offline-sqs

This might be related to the general discussion in https://github.com/CoorpAcademy/serverless-plugins/issues/102

I created a very minimal demo here: https://github.com/zoellner/offline-sqs-bug

The issue comes down to a very simple piece of code:
```js
const AWS = require('aws-sdk');

exports.handler = async function handleEvent(event) {
  console.log(event);
  return true;
}
```

Triggering two messages to this code will throw an error (the first one runs fine). The problem is in the variable require that's outside the event handler. 
The demo with the aws-sdk is just an example since that's a very common dependency. I've seen a similar issue using pino logger and others.

**Comment by benhutchins** [comment]
Might want to check https://github.com/dherault/serverless-offline/issues/1075 to see if that is the issue you're having.

**Comment by zoellner** [comment]
Yes, I think it comes down to the same underlying change in serverless-offline (was a bit too far into the weeds to realize when I opened this ticket and forgot to link it)

But the proposed workarounds for serverless-offline api endpoints (`useChildProcesses` and `allowCache`) don't work in the  combination with serverless-offline-sqs. I'm assuming that there's something in the way the plugin calls the lambdas that ignores some of these settings.

**Comment by sambP** [comment]
for a quick fix, this combination worked for me:
    "serverless-offline": "5.12.1",
    "serverless-offline-sqs": "3.1.3",

**Comment by silouone** [comment]
Thanks for the minimal repro @zoellner — that's exactly what makes these tractable. The serverless-offline compatibility and message-handling rework in #265 (on master now, released shortly) overhauls how `serverless-offline-sqs` invokes handlers, so the second-message failure with module-level requires should be addressed; would you mind re-testing your demo against master or the next release and reporting back? 🙏

**Comment by silouone** [comment]
Thanks for the minimal repro @zoellner — the `offline-sqs-bug` demo made this one easy to pin down, and you were right that it's the same underlying serverless-offline change you flagged from #102.

The root cause was that the plugin used to drive invocation/module-loading in a way that didn't honor serverless-offline's handler caching, so a second message re-evaluated module-scope state (your `require('aws-sdk')`, pino, etc.) and blew up — and that's also why `useChildProcesses`/`allowCache` didn't help.

This is **fixed on master** by #265: the plugin no longer runs handlers itself. It now delegates entirely to serverless-offline's own Lambda runner — `index.js` dynamically imports `serverless-offline/lambda` and `sqs.js` calls `lambdaFunction.runHandler()` — so module-scope state is preserved across invocations exactly like AWS. It's pinned against serverless-offline 13.

It's fixed on master and will ship in the next release of `serverless-offline-sqs` (not yet on npm). Closing as completed — please reopen if your repro still fails after you upgrade to the next release. 🙏
